src/utils.ts

```diff
@@ -914,7 +914,7 @@ export function makeInterruptibleAsyncInterval(
   }
 
   function executeAndReschedule() {
-    cannotBeExpedited = false;
+    lastWakeTime = 0;
```

Comment:
Are we sure that this is the right reset behavior? Why do we need to reset the lastWakeTime at all? It's set to currentTime by the wake function, which is what calls the execute function here, so it should be correct. The previous code was manually setting a flag to allow or forbid expediting the wake, so it needed to be reset. What is the reasoning for resetting it here?

Actually, looking over the code some more, it seems that a cleaner approach might be to leave the `cannotBeExpedited` logic as is, but add a line at the top of `wake` to override the value to `true` if the lastWakeTime is negative, i.e. `cannotBeExpedited = cannotBeExpedited ||  lastWakeTime < 0`. I think it would make it easier to reason about what exactly we are resetting and why.

Reply:
I agree the `cannotBeExpedited` logic was easier to reason with but I am hesitant to bring it back for 2 reasons: 1) I could not replicate the exact same behaviour in my lambda testing that the customer was seeing and 2) I believe that `cannotBeExpedited` logic was potentially wrong. In the previous implementation it looked possible to continuously "wake" with a negative `timeUntilNextCall` (some lambda situations) that would never debounce the calls. This revert to the old time based logic would at least ensure it could not be woken continuously sooner than the minimum interval.